### PR TITLE
Add index relationship to case mock

### PR DESCRIPTION
**Pull request by proteusvacuum**
I think a `CaseRelationship` in this mock is actually known as a `CaseIndex` elsewhere. @czue, do you think this is worth changing, or is there a reason you called it a `CaseRelationship` that I'm not seeing?(https://github.com/dimagi/commcare-hq/pull/6147)

Either way, this extra line needs to get added for testing cases with different index relationships (i.e. extensions).

@sravfeyn

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/case/tests/test_factory.py [hunk 1]
```diff
@@ -24,12 +24,13 @@ class CaseStructureTest(SimpleTestCase):
...
-            relationships=[
-                CaseRelationship(CaseStructure(case_id=parent_case_id))
+            indices=[
+                CaseIndex(CaseStructure(case_id=parent_case_id))
             ]
         )
         self.assertEqual(
-            {CaseRelationship.DEFAULT_RELATIONSHIP: (CaseRelationship.DEFAULT_RELATED_CASE_TYPE, parent_case_id)},
+            {CaseIndex.DEFAULT_RELATIONSHIP: (CaseIndex.DEFAULT_RELATED_CASE_TYPE, parent_case_id,
+                                                     CaseIndex.DEFAULT_RELATIONSHIP)},
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/tests/test_factory.py]: E127 continuation line over-indented for visual indent

**Comment by proteusvacuum** [comment]
I ended up changing it... Mostly so that I could write a test where there was more than one index of the same type.

Wait for tests, if they ever start (!)...

**Comment by czue** [comment]
i don't remember why i called it relationship. possibly so that it was clearer what type of object I was working with (sometimes searching for things like `index` or `indices` in the code gets hard)? 

happy with this change if you find it more clear. agree we should wait for tests.

File: corehq/ex-submodules/casexml/apps/case/mock.py [hunk 2]
```diff
@@ -278,38 +280,42 @@ class CaseStructure(object):
...
 
 
-class CaseRelationship(object):
-    DEFAULT_RELATIONSHIP = 'parent'
+class CaseIndex(object):
+    DEFAULT_RELATIONSHIP = 'child'
     DEFAULT_RELATED_CASE_TYPE = 'default_related_case_type'
 
-    def __init__(self, related_structure=None, relationship=DEFAULT_RELATIONSHIP, related_type=None):
+    def __init__(self, related_structure=None, relationship=DEFAULT_RELATIONSHIP, related_type=None, identifier=None):
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/mock.py]: E501 line too long (118 > 115 characters)

**Comment by czue** [comment]
:+1: to merge when tests pass - feel free to ping or self-merge then

**Comment by proteusvacuum** [comment]
A better way of doing this, https://github.com/dimagi/commcare-hq/pull/8600

**Pull request closed without merging**